src/lib/broadcast.jl

```diff
@@ -45,18 +45,20 @@ function Base.reducedim_init(::typeof(identity), ::typeof(accum), A::AbstractArr
   Base.reducedim_initarray(A, region, nothing, Union{Nothing,eltype(A)})
 end
 
-trim(x, Δ) = reshape(Δ, ntuple(i -> size(Δ, i), Val(ndims(x))))
-trim(x::Tuple, Δ) = NTuple{length(x)}(Δ)
-
-unbroadcast(x::AbstractArray, x̄) =
-  size(x) == size(x̄) ? x̄ :
-  length(x) == length(x̄) ? trim(x, x̄) :
-    trim(x, accum_sum(x̄, dims = ntuple(i -> size(x, i) == 1 ? i : ndims(x̄)+1, Val(ndims(x̄)))))
-
+function unbroadcast(x::AbstractArray, x̄)
+  N = ndims(x̄)
+  if length(x) == length(x̄)
+    _project(x, x̄)  # ProjectTo handles reshape, offsets, structured matrices, row vectors
+  else
+    tup = filter(d -> size(x, d) == 1, ntuple(identity, N))
+    dims = length(tup) == 1 ? only(tup) : tup  # avoid sum(xbar, dims=(1,)) as e.g. sum(SA[1 2; 3 4], dims=(1,)) fails
```

Comment:
Is there a downside to passing the tuple?

Reply:
Did you try the example provided on precisely the line you quoted?

Reply:
Oh the comment goes beyond the dims. Still, that seems like a SA issue. Do they have a corresponding issue to support the syntax? Do you have a link?

Reply:
Found https://github.com/JuliaArrays/StaticArrays.jl/pull/909

Reply:
Yeah, so we can avoid this line in that case and SA can make a release that fixes the bug.